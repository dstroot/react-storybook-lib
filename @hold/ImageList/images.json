[
  {
    "id": "1",
    "date": "11-14-18",
    "full": "/media/wildfire.jpg",
    "thumb": "/media/wildfire.jpg",
    "title": "California Wildfire",
    "caption": "",
    "alt": "wildfire",
    "description": "Pacific Life Foundation Donates $250,000 to the American Red Cross for Victims of the California Wildfires",
    "category": "Logos"
  },
  {
    "id": "2",
    "date": "11-14-18",
    "alt": "students",
    "full": "/media/students.jpg",
    "thumb": "/media/students.jpg",
    "title": "Students",
    "description": "Pacific Life Foundation Donates $250,000 to the American Red Cross for Victims of the California Wildfires",
    "category": "Logos"
  },
  {
    "id": "3",
    "date": "11-14-18",
    "alt": "awards",
    "full": "/media/awards.jpg",
    "thumb": "/media/awards.jpg",
    "title": "Awards",
    "description": "Pacific Life Foundation Donates $250,000 to the American Red Cross for Victims of the California Wildfires",
    "category": "Logos"
  }
]
