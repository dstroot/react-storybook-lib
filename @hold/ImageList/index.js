import React from "react";
import PropTypes from "prop-types";

// components
import ImageItem from "../../components/ImageItem";

const ImageList = ({ data }) => {
  return (
    <div className="container mt-3">
      <div className="row">
        <div className="col">
          <h1 className="display-6 mb-2 mt-4">All Logos</h1>
        </div>
      </div>
      <div className="row">
        {data.map(item => (
          <ImageItem image={item} key={item.id} />
        ))}
      </div>
      <p className="small text-muted mt-4">
        Images on this page may be used for publication with credit: "Source:
        Pacific Life."
      </p>
    </div>
  );
};

export default ImageList;

// define props
ImageList.propTypes = {
  data: PropTypes.array.isRequired
};
