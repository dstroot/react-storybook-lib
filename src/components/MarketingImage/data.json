[
  {
    "image": "/media/1.jpg",
    "lead": "Plan now, so you’re ready later",
    "text": "For you, family is one of the most important things in your life. You take care of them. They take care of you. Making sure that they’ll always be taken care of, no matter what happens.",
    "url": "/"
  },

  {
    "image": "/media/2.jpg",
    "lead": "Save enough for my future",
    "text": "You don’t know what the future has planned for you, but you want to be prepared for the unexpected and be able to achieve your goals.",
    "url": "/"
  },
  {
    "image": "/media/3.jpg",
    "lead": "Protect and grow my business",
    "text": "You want your business to maintain its competitive edge. Attracting talent and building a succession plan for the future means you can ensure your business stays in stable hands.",
    "url": "/"
  }
]
