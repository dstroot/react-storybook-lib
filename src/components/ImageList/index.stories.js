import React from "react";
import { storiesOf } from "@storybook/react";

// components
import ImageList from "../ImageList";
import data from "./images.json";

// ImageList section
storiesOf("ImageList", module).add("show ImageList", () => (
  <ImageList data={data} />
));
