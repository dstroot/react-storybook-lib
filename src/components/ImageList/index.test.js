import React from "react";
import renderer from "react-test-renderer";

import ImageList from "../ImageList";
import data from "./images.json";

describe("ImageList", () => {
  it("it should render", () => {
    const component = renderer.create(<ImageList data={data} />);
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
