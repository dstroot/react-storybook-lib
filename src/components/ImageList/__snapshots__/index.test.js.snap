// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ImageList it should render 1`] = `
<div
  className="container mt-3"
>
  <div
    className="row"
  >
    <div
      className="col"
    >
      <h1
        className="display-6 mb-2 mt-4"
      >
        All Logos
      </h1>
    </div>
  </div>
  <div
    className="row"
  >
    <div
      className="col-sm-4 mt-4"
    >
      <img
        alt="wildfire"
        className="img-fluid link"
        height="400px"
        onClick={[Function]}
        src="/media/wildfire.jpg"
        width="400px"
      />
      <br />
      <p
        className="lead d-inline"
      >
        <a
          aria-label="Download Google"
          className=""
          download={true}
          href="/media/wildfire.jpg"
        >
          <img
            alt="arrow"
            className="mr-2"
            src="download-arrow.svg"
            width="18px"
          />
        </a>
        California Wildfire
      </p>
    </div>
    <div
      className="col-sm-4 mt-4"
    >
      <img
        alt="students"
        className="img-fluid link"
        height="400px"
        onClick={[Function]}
        src="/media/students.jpg"
        width="400px"
      />
      <br />
      <p
        className="lead d-inline"
      >
        <a
          aria-label="Download Google"
          className=""
          download={true}
          href="/media/students.jpg"
        >
          <img
            alt="arrow"
            className="mr-2"
            src="download-arrow.svg"
            width="18px"
          />
        </a>
        Students
      </p>
    </div>
    <div
      className="col-sm-4 mt-4"
    >
      <img
        alt="awards"
        className="img-fluid link"
        height="400px"
        onClick={[Function]}
        src="/media/awards.jpg"
        width="400px"
      />
      <br />
      <p
        className="lead d-inline"
      >
        <a
          aria-label="Download Google"
          className=""
          download={true}
          href="/media/awards.jpg"
        >
          <img
            alt="arrow"
            className="mr-2"
            src="download-arrow.svg"
            width="18px"
          />
        </a>
        Awards
      </p>
    </div>
  </div>
  <p
    className="small text-muted mt-4"
  >
    Images on this page may be used for publication with credit: "Source: Pacific Life."
  </p>
</div>
`;
